src/node_crypto.cc

```diff
@@ -3478,29 +3478,28 @@ void CipherBase::Init(const char* cipher_type,
                                key,
                                iv);
 
-  EVP_CIPHER_CTX_init(&ctx_);
+  ctx_ = EVP_CIPHER_CTX_new();
```

Comment:
The extra heap allocation is the thing that porting to OpenSSL 1.1.0 requires. The structs are hidden, which means they cannot be stack-allocated. Alas, that does mean there are two allocations total (these objects internally allocate). It could have been one had upstream not messed up their API (`EVP_CIPHER_CTX_new` and `EVP_EncryptInit` should have been merged into one function), but so it goes. :-/

(`EVP_CIPHER_CTX_reset` is a basically useless function. The original PR used it, but it was wrong and broke the 1.0.2 build. init and reset do very different things.)